src/lib/features/project-status/projects-status.e2e.test.ts

```diff
@@ -21,20 +29,67 @@ beforeAll(async () => {
         },
         db.rawDatabase,
     );
+    eventService = createEventsService(db.rawDatabase, config);
 });
 
 afterAll(async () => {
     await app.destroy();
     await db.destroy();
 });
 
-test('project insights happy path', async () => {
+test('project insights should return correct count for each day', async () => {
+    await eventService.storeEvent({
+        type: FEATURE_CREATED,
+        project: 'default',
+        data: { featureName: 'today-event' },
+        createdBy: 'test-user',
+        createdByUserId: TEST_USER_ID,
+        ip: '127.0.0.1',
+    });
+
+    await eventService.storeEvent({
+        type: FEATURE_CREATED,
+        project: 'default',
+        data: { featureName: 'today-event-two' },
+        createdBy: 'test-user',
+        createdByUserId: TEST_USER_ID,
+        ip: '127.0.0.1',
+    });
+
+    await eventService.storeEvent({
+        type: FEATURE_CREATED,
+        project: 'default',
+        data: { featureName: 'yesterday-event' },
+        createdBy: 'test-user',
+        createdByUserId: TEST_USER_ID,
+        ip: '127.0.0.1',
+    });
+
+    const { events } = await eventService.getEvents();
+
+    const lateEvent = events.find(
+        (e) => e.data.featureName === 'yesterday-event',
+    );
+    await db.rawDatabase.raw(
+        `UPDATE events SET created_at = created_at - interval '1 day' where id = ?`,
+        [lateEvent?.id],
+    );
+
     const { body } = await app.request
         .get('/api/admin/projects/default/status')
         .expect('Content-Type', /json/)
         .expect(200);
 
+    const today = new Date();
+    const todayString = today.toISOString().split('T')[0];
+    const yesterday = new Date(today);
+    yesterday.setDate(today.getDate() - 1);
+    const yesterdayString = yesterday.toISOString().split('T')[0];
```

Comment:
that's a lot of date logic for a test. Either make it work with hardcoded value or extract method refactoring to keep the test readable and focused